src/portal/components/navbar/navbar.component.ts

```diff
@@ -124,5 +128,18 @@ export const NavbarComponent: ng.IComponentOptions = {
         vm.tasksScheduler = undefined;
       }
     }
+
+    vm.authenticate = function(provider: string) {
+      AuthenticationService.authenticate(provider)
+        .then( () => {
+          UserService.current().then( () => {
+            vm.$rootScope.$broadcast('graviteeUserRefresh');
+          });
+        })
+        .catch( () => {});
+    };
+
+    vm.isOnlyOAuth = !Constants.localLoginEnabled && vm.providers.length == 1;
```

Comment:
Does it means that if social authentication is enabled, it is not under the `oauth` scope ?

Reply:
it is only a way to compute the fact that local login is disable and that there is only one single identity provider (it does not really matter if generic oauth2 or google ect). 
in this case we set the link to the external provider on the top right, if local login is enabled or there are more than one external identity provider, we open the form where the user can choose between the list of external identity providers.